api/src/controllers/topics.js

```diff
@@ -0,0 +1,57 @@
+import { github } from '../config/index.json';
+import Ajv from 'ajv';
+import schema from '../schemas/topic.json';
+import { randomId } from '../utils/strings';
+import {
+  openPullExistsForBranch,
+  createNewRefFromBase,
+  createFile,
+  createPullRequest,
+} from '../utils/github';
+import slugify from 'slugify';
+
+const ajv = new Ajv();
+
+const validate = ajv.compile(schema);
+
+export const createOrUpdateTopic = async (req, res) => {
+  const branchName = `${github.branchPrefix}/${randomId(github.branchIdLength)}`;
+  const { repo, owner, defaultBranch, email, name } = github;
+  let status = '200';
+  let statusMessage = 'Ok';
+  const bodyData = JSON.stringify(req.body, null, 2);
+  const topicName = slugify(req.body.name.toLowerCase(), '-');
+
+  const ref = `refs/heads/createTopic/${topicName}`;
+
+  try {
+    if (await openPullExistsForBranch(branchName, repo, owner)) {
+      // add payload to pull request as new commit
+    } else {
+      // create new topic and branch
+      // validate topic schema
+      const isValidData = validate(req.body);
+      if (isValidData) {
+        // TODO -- validate topic doesn't already exist
+        // TODO -- validate topic sources are valid
+        // create a git branch on the remote with a naming convention [createTopic/topicname]
+        await createNewRefFromBase(owner, repo, ref);
+        // // create a new file with contents
+        await createFile(owner, repo, bodyData, ref, topicName, email, name);
+        // commit  to branch
+        // make pr against ref to base using templates
+        const pullRequest = await createPullRequest(owner, repo, defaultBranch, topicName, ref);
+        // URL to the pull Request created ..
+        const pullRequest_url = pullRequest.data.html_url;
+        statusMessage = `Pull Request Created at ${pullRequest_url}`;
+      } else {
+        status = '400';
+        statusMessage = 'Bad Request';
+      }
+    }
+  } catch (e) {
+    status = e.status;
+    statusMessage = e;
+  }
+  res.status(status).send(statusMessage);
```

Comment:
this may work for now but because we are sending json data forward, it is  pretty common to get JSON back. The JSON payload has the info in it. This way it is also organized. m

`res.json({prUrl: ...})`;

its better to simplify then make oppinions on the backend. If you send the PR URL the front end can represent it in anyway it wants. 

What you do want to do is at good console.logs so that the pods stdout has an audit of PR's being created.